pupil_src/shared_modules/surface_tracker/surface_tracker_offline.py

```diff
@@ -0,0 +1,785 @@
+                pyglui_utils.draw_polyline(
+                    surface, color=color, line_type=gl.GL_LINES, thickness=scale * 2
+                )
+
+    def _draw_labels(self, width, height, scale):
+        self.glfont.set_size(self.timeline_line_height * 0.8 * scale)
+        self.glfont.draw_text(width, 0, "Marker Cache")
+        for idx, s in enumerate(self.surfaces):
+            gl.glTranslatef(0, self.timeline_line_height * scale, 0)
+            self.glfont.draw_text(width, 0, s.name)
+
+    def add_surface(self, _=None, init_dict=None):
+        super().add_surface(init_dict=init_dict)
+        # Plugin initialization loads surface definitions before UI is initialized. Changing timeline height will fail in this case.
+        if self.markers or init_dict is not None:
+            try:
+                self.timeline.content_height += self.timeline_line_height
+                self._fill_gaze_on_surf_buffer()
+            except AttributeError:
+                pass
+            self.surfaces[-1].on_surface_changed = self.on_surface_change
+
+    def remove_surface(self, _):
+        super().remove_surface(_)
+        self.timeline.content_height -= self.timeline_line_height
+
+    def on_notify(self, notification):
+        super().on_notify(notification)
+
+        if notification["subject"] == "surface_tracker.marker_detection_params_changed":
+            self._recalculate_marker_cache()
+
+        elif notification["subject"] == "surface_tracker.marker_min_perimeter_changed":
+            self._update_filtered_markers()
+            for surface in self.surfaces:
+                surface.location_cache = None
+
+        elif notification["subject"] == "surface_tracker.heatmap_params_changed":
+            for surface in self.surfaces:
+                if surface.name == notification["name"]:
+                    self._heatmap_update_requests.add(surface)
+                    surface.within_surface_heatmap = surface.get_placeholder_heatmap()
+                    break
+            self._fill_gaze_on_surf_buffer()
+
+        elif notification["subject"].startswith("seek_control.trim_indices_changed"):
+            for surface in self.surfaces:
+                surface.within_surface_heatmap = surface.get_placeholder_heatmap()
+                self._heatmap_update_requests.add(surface)
+            self._fill_gaze_on_surf_buffer()
+
+        elif notification["subject"] == "surface_tracker.surfaces_changed":
+            for surface in self.surfaces:
+                if surface.name == notification["name"]:
+                    surface.location_cache = None
+                    surface.within_surface_heatmap = surface.get_placeholder_heatmap()
+                    self._heatmap_update_requests.add(surface)
+                    break
+            self._fill_gaze_on_surf_buffer()
+
+        elif notification["subject"] == "should_export":
+            self.make_export = True
```

Comment:
Let's discuss how the export is handled in person, maybe we find a solution that does not rely on these trigger variables.